execution/aggregate/hashaggregate.go

```diff
@@ -130,7 +130,10 @@ func (a *aggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 		return nil, err
 	}
 	for i := range args {
-		a.params[i] = args[i].Samples[0]
```

Comment:
it looks like we are fixing a defect with "scalar" here; we should fix the root-cause why args has no samples rather

Reply:
Hm so will this always come from `scalar` now? Tests started to fail once we made `noop` return empty vectors.

Reply:
I updated the PR to not pass `noop` into the aggregate and no we check for `nil` explicitly.